controllers/component_build_controller_pac.go

```diff
@@ -908,15 +692,14 @@ func generatePaCPipelineRunForComponent(
 		{Name: "output-image", Value: tektonapi.ArrayOrString{Type: "string", StringVal: proposedImage}},
 	}
 	if onPull {
-		expiration := os.Getenv(pipelineRunOnPRExpirationEnvVar)
-		validExpiration, _ := regexp.Match("^[1-9][0-9]{0,2}[hdw]$", []byte(expiration))
-		if !validExpiration {
-			if expiration != "" {
-				log.Info(fmt.Sprintf("invalid expiration '%s' in %s envVar, using default %s", expiration, pipelineRunOnPRExpirationEnvVar, pipelineRunOnPRExpirationDefault), l.Action, l.ActionAdd)
+		prImageExpiration := PipelineRunOnPRExpirationDefault
```

Comment:
not sure why we have another validation of PR image expiration here, since it was already checked in `main.go`?

Reply:
Validation in main was added to instantly inform about a problem with the configuration, where as validation in the code above was for actual using.
@psturc your comment made me rethink the approach. It looks like we can eliminate invalid configuration in main by overriding the environment variable. So, the controller expects only valid or empty value.